test/optimizer/column_stats_test.cpp

```diff
@@ -0,0 +1,118 @@
+#include "common/harness.h"
+
+#include <vector>
+
+#include "common/logger.h"
+#include "optimizer/stats/column_stats.h"
+#include "type/type.h"
+#include "type/value.h"
+#include "type/value_factory.h"
+
+#define private public
+#define TEST_OID 0
+
+namespace peloton {
+namespace test {
+
+using namespace optimizer;
+
+class ColumnStatsTests : public PelotonTest {};
+
+using ValueFrequencyPair = std::pair<type::Value, double>;
+
+void PrintCommonValueAndFrequency(
+    std::vector<std::pair<type::Value, double>> valfreq) {
+  for (auto const& p : valfreq) {
+    LOG_INFO("\n [Print k Values] %s, %f", p.first.GetInfo().c_str(), p.second);
+  }
+}
+
+TEST_F(ColumnStatsTests, BasicTest) {
+  ColumnStats colstats{TEST_OID, TEST_OID, TEST_OID,
+                       type::Type::TypeId::INTEGER};
+  for (int i = 0; i < 10; i++) {
+    type::Value v = type::ValueFactory::GetIntegerValue(i);
+    colstats.AddValue(v);
+  }
+  EXPECT_EQ(colstats.GetCardinality(), 10);
+  EXPECT_EQ(colstats.GetFracNull(), 0);
+  EXPECT_EQ(colstats.GetHistogramBound().size() + 1, colstats.num_bins);
+  EXPECT_EQ(colstats.GetCommonValueAndFrequency().size(), colstats.top_k);
+}
+
+// Test categorical values
+TEST_F(ColumnStatsTests, DistinctValueTest) {
+  ColumnStats colstats{TEST_OID, TEST_OID, TEST_OID,
+                       type::Type::TypeId::BOOLEAN};
+  for (int i = 0; i < 1250; i++) {
+    type::Value v = type::ValueFactory::GetBooleanValue(i % 5 == 0);
+    colstats.AddValue(v);
+  }
+  EXPECT_EQ(colstats.GetCardinality(), 2);
+  EXPECT_EQ(colstats.GetFracNull(), 0);
+  EXPECT_EQ(colstats.GetHistogramBound().size(),
+            0);  // No histogram for categorical data
+  std::vector<ValueFrequencyPair> valfreq =
+      colstats.GetCommonValueAndFrequency();
+  // EXPECT_EQ(valfreq.size(), 2);
+  // PrintCommonValueAndFrequency(colstats.GetCommonValueAndFrequency());
+}
+
+// All stats collectors should work with trivial case
+TEST_F(ColumnStatsTests, TrivialValueTest) {
```

Comment:
I think this unit test case can be removed later on. For now, this may be fine from a code development and testing perspective.